ibis/pyspark/compiler.py

```diff
@@ -1035,6 +1038,26 @@ def compile_join(t, expr, scope, timecontext, *, how):
     return left_df.join(right_df, pred_columns, how)
 
 
+def _canonicalize_interval(interval):
+    """ Convert interval to integer timestamp of second
+
+    When pyspark cast timestamp to integer type, it uses the number of seconds
+    since epoch. Therefore, we need cast ibis interval correspondingly.
+    """
+
+    if isinstance(interval, ir.IntervalScalar):
+
+        return int(execute(interval).value / 1e9)
```

Comment:
is there a concern if this is too big here? e.g. i think resolution on spark is only ms?

Reply:
Here in window bounds spark uses seconds since epoch, execute gives us a Timedelta and value is in nanosecond